README.md

```diff
@@ -50,6 +50,42 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described
+[here](https://sanctuary.js.org/#types). In summary:
```

Comment:
@alex-deas - personally, I'd link to anything I might think useful to people without prior knowledge of Haskell. The type signature notation is based on Haskell's - with (AFAIK) some extensions since we're annotating JS code (like the `~>` for "method type constructor").

I am currently re-writing this PR and think I'll make the link to Sanctuary less prominent (i.e. footnote - "for more, see Link")